src/service.rs

```diff
@@ -634,31 +579,35 @@ where
                         .await;
                     });
                     self.session_wait_handle
-                        .insert((id, proto_id), (Some(sender), handle));
+                        .insert((id, *proto_id), (Some(sender), handle));
                 }
+            } else {
+                debug!("can't find proto [{}] session handle", proto_id);
             }
         }
     }
 
     fn push_session_message(
-        &mut self,
-        session_id: SessionId,
-        proto_id: ProtocolId,
+        ctx: &SessionController,
         priority: Priority,
+        proto_id: ProtocolId,
+        high_write_buf: &mut VecDeque<(SessionId, SessionEvent)>,
```

Comment:
I think this fn can be simplified to a helper fn without `Priority` parameter:

```
push_session_message(
  ctx: &SessionController,
  proto_id: ProtocolId,
  buf: &mut VecDeque<(SessionId, SessionEvent)>,
  data: Bytes,
) {
  
}
```

in invoker code, check the priority first:

```
let buf = match priority {
  High => self.high_write_buf,
  Normal => self.write_buf,
};

match target {
  ...
  Service::<()>::push_session_message(
                        control,
                        proto_id,
                        &buf,
                        data,
                    )
  ...
}
```

Reply:
The Priority parameter is currently required and can be removed here #240